src/main/java/games/powergrid/PowerGridHeuristic.java

```diff
@@ -0,0 +1,167 @@
+package games.powergrid;
+
+
+import java.util.Map;
+
+import core.AbstractGameState;
+import core.components.Deck;
+import core.interfaces.IStateHeuristic;
+import games.powergrid.PowerGridParameters.Resource;
+import games.powergrid.components.PowerGridCard;
+import games.powergrid.components.PowerGridCard.PlantInput;
+
+public class PowerGridHeuristic implements IStateHeuristic {
+
+	/**
+	 * Evaluates the current {@link PowerGridGameState} from the perspective of the specified player
+	 * using a leader-based heuristic that compares the player's performance against the current leader
+	 * across multiple key categories.
+	 * <p>
+	 * This heuristic is designed for intermediate or non-terminal game states in Power Grid.
+	 * It returns a normalized score in the range [0, 1], representing how well the player is doing
+	 * relative to the top-performing players in the following categories:
+	 * <ul>
+	 *   <li>Number of cities controlled</li>
+	 *   <li>Money (cash on hand)</li>
+	 *   <li>Income (turn-based earnings)</li>
+	 *   <li>Power plant capacity</li>
+	 *   <li>Resource reserves vs. required fuel</li>
+	 * </ul>
+	 * Each component is scaled, weighted, and combined into a single heuristic value.
+	 * If the game is over, this function returns {@code 10.0} for a win and {@code 0.0} otherwise.
+	 * <p>
+	 * The function relies on helper methods like {@code getMaxExcluding()} to determine the
+	 * leader in each category (excluding the current player) and {@code shapedGap()} to apply
+	 * a nonlinear scaling that rewards proximity to or exceeding the leader's value.
+	 *
+	 * @param gs        the current abstract game state, expected to be an instance of {@link PowerGridGameState}
+	 * @param playerId  the ID of the player whose perspective the state is being evaluated from
+	 * @return a normalized heuristic value in [0, 1] representing the player’s standing compared to the current leaders;
+	 *         returns {@code 10.0} if the player is the winner, {@code 0.0} if the player has lost
+	 *
+	 * @see PowerGridGameState
+	 * @see #getMaxExcluding(int[], int)
+	 * @see #shapedGap(double, double, double, double, double, double)
+	 */
+
+	@Override
+	public double evaluateState(AbstractGameState gs, int playerId) {
+	    PowerGridGameState s = (PowerGridGameState) gs;
+	    if (s.isGameOver()) {
+	        return (gs.getWinner() == playerId) ? 1.0 : -1.0;
```

Comment:
This fails to compile - you want to use gs.getWinners(), which returns a Set of integers, allowing for draws.

(or actually better, check GameResult for the player)